libqtile/widget/windowtabs.py

```diff
@@ -48,6 +50,7 @@ class WindowTabs(base._TextBox):
     ]
 
     def __init__(self, **config):
+        warnings.warn("Use tasklist(interactive=False) instead", DeprecationWarning)
```

Comment:
Why not just 

```suggestion
        warnings.warn("Use tasklist(mouse_callbacks=None) instead", DeprecationWarning)
```

and not add an extra option for this?

Reply:
That does make more sense. I had to add a check to the button_press function so it doesn't run and throw errors when there are no callbacks. I also added a tasklist docstring for mouse_callbacks that overrides the original and explains setting the value to none.

Reply:
Uh I thought None was already supported for all widgets. If not we should probably do this instead of here only.

Reply:
Also I don't understand why we need the extra check and that seems weird... but I need to sleep right now. :)

Reply:
have a good night.
When I set `mouse_callbacks=None` I get this error
```
Traceback (most recent call last):
  File "/home/yobleck/qtile_dev/update_tasklist/qtile/libqtile/backend/x11/core.py", line 298, in _xpoll
    ret = target(event)
  File "/home/yobleck/qtile_dev/update_tasklist/qtile/libqtile/backend/x11/window.py", line 1070, in handle_ButtonPress
    self.process_button_click(e.event_x, e.event_y, e.detail)
  File "/home/yobleck/qtile_dev/update_tasklist/qtile/libqtile/bar.py", line 362, in process_button_click
    widget.button_press(
  File "/home/yobleck/qtile_dev/update_tasklist/qtile/libqtile/widget/tasklist.py", line 430, in button_press
    base._Widget.button_press(self, x, y, button)
  File "/home/yobleck/qtile_dev/update_tasklist/qtile/libqtile/widget/base.py", line 236, in button_press
    if name in self.mouse_callbacks:
TypeError: argument of type 'NoneType' is not iterable
```

So it seems that `None` isn't supported for the mouse_callback argument?
perhaps changing :https://github.com/qtile/qtile/blob/e928649da5919f30cfa4db9348f51e7d2e851e7b/libqtile/widget/base.py#L236
to
`if self.mouse_callbacks and name in self.mouse_callbacks:`
would be a solution (it fixes the error but IDK if it's the right way to do it).

Reply:
Yeah, that would work I think!